### ISSUE TITLE
Issues with installation of MESSAGEix using `pip` 

### ISSUE THREAD

**Issue by OFR-IIASA** [iiasa/message_ix#468]
There is an issue occurring when installing MESSAGEix using `pip install` with the latest version of pip.

`WARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617`

The extended error message as follows:

```
WARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617>
distutils: c:\users\fricko\anaconda3\Include\UNKNOWN
sysconfig: c:\users\fricko\anaconda3\Include
WARNING: Additional context:
user = False
home = None
root = None
prefix = None
WARNING: Ignoring invalid distribution -xmp (c:\users\fricko\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -pype1 (c:\users\fricko\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -xmp (c:\users\fricko\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -pype1 (c:\users\fricko\anaconda3\lib\site-packages)
Obtaining file:///C:/Repo/message_ix
  WARNING: Ignoring invalid distribution -xmp (c:\users\fricko\anaconda3\lib\site-packages)
  WARNING: Ignoring invalid distribution -pype1 (c:\users\fricko\anaconda3\lib\site-packages)
Requirement already satisfied: click in c:\users\fricko\anaconda3\lib\site-packages (from message-ix==3.2.1.dev67+ga20ffb0.d20210427) (7.1.2)
Requirement already satisfied: ixmp>=3.2.0 in c:\repo\ixmp (from message-ix==3.2.1.dev67+ga20ffb0.d20210427) (3.2.1.dev80+g40fc589)
Requirement already satisfied: numpy in c:\users\fricko\anaconda3\lib\site-packages (from message-ix==3.2.1.dev67+ga20ffb0.d20210427) (1.19.2)
Requirement already satisfied: pandas in c:\users\fricko\anaconda3\lib\site-packages (from message-ix==3.2.1.dev67+ga20ffb0.d20210427) (1.1.5)
Requirement already satisfied: PyYAML in c:\users\fricko\anaconda3\lib\site-packages (from message-ix==3.2.1.dev67+ga20ffb0.d20210427) (5.3.1)
Requirement already satisfied: setuptools>=41 in c:\users\fricko\anaconda3\lib\site-packages (from message-ix==3.2.1.dev67+ga20ffb0.d20210427) (51.0.0.post20201207)
Requirement already satisfied: genno>=1.1.0 in c:\users\fricko\anaconda3\lib\site-packages (from ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (1.3.0)
Requirement already satisfied: JPype1>=1.2.1 in c:\users\fricko\anaconda3\lib\site-packages (from ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (1.2.1)
Requirement already satisfied: openpyxl in c:\users\fricko\anaconda3\lib\site-packages (from ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (3.0.5)
Requirement already satisfied: pint in c:\users\fricko\anaconda3\lib\site-packages (from ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (0.11)
Requirement already satisfied: xarray in c:\users\fricko\anaconda3\lib\site-packages (from ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (0.15.1)
Requirement already satisfied: dask[array]>=2.14 in c:\users\fricko\anaconda3\lib\site-packages (from genno>=1.1.0->ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (2020.12.0)
Requirement already satisfied: sparse>=0.12 in c:\users\fricko\anaconda3\lib\site-packages (from genno>=1.1.0->ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (0.12.0)
Requirement already satisfied: toolz>=0.8.2 in c:\users\fricko\anaconda3\lib\site-packages (from dask[array]>=2.14->genno>=1.1.0->ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (0.11.1)
Requirement already satisfied: typing-extensions in c:\users\fricko\anaconda3\lib\site-packages (from JPype1>=1.2.1->ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (3.7.4.3)
Requirement already satisfied: python-dateutil>=2.7.3 in c:\users\fricko\anaconda3\lib\site-packages (from pandas->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (2.8.1)
Requirement already satisfied: pytz>=2017.2 in c:\users\fricko\anaconda3\lib\site-packages (from pandas->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (2020.4)
Requirement already satisfied: six>=1.5 in c:\users\fricko\anaconda3\lib\site-packages (from python-dateutil>=2.7.3->pandas->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (1.15.0)
Requirement already satisfied: scipy>=0.19 in c:\users\fricko\anaconda3\lib\site-packages (from sparse>=0.12->genno>=1.1.0->ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (1.5.2)
Requirement already satisfied: numba>=0.49 in c:\users\fricko\anaconda3\lib\site-packages (from sparse>=0.12->genno>=1.1.0->ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (0.51.2)
Requirement already satisfied: llvmlite<0.35,>=0.34.0.dev0 in c:\users\fricko\anaconda3\lib\site-packages (from numba>=0.49->sparse>=0.12->genno>=1.1.0->ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (0.34.0)
Requirement already satisfied: jdcal in c:\users\fricko\anaconda3\lib\site-packages (from openpyxl->ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (1.4.1)
Requirement already satisfied: et-xmlfile in c:\users\fricko\anaconda3\lib\site-packages (from openpyxl->ixmp>=3.2.0->message-ix==3.2.1.dev67+ga20ffb0.d20210427) (1.0.1)
WARNING: Ignoring invalid distribution -xmp (c:\users\fricko\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -pype1 (c:\users\fricko\anaconda3\lib\site-packages)
Installing collected packages: message-ix
  Attempting uninstall: message-ix
    WARNING: Ignoring invalid distribution -xmp (c:\users\fricko\anaconda3\lib\site-packages)
    WARNING: Ignoring invalid distribution -pype1 (c:\users\fricko\anaconda3\lib\site-packages)
    Found existing installation: message-ix 3.3.0a1.dev11+gac140c6.d20210423
    Uninstalling message-ix-3.3.0a1.dev11+gac140c6.d20210423:
      Successfully uninstalled message-ix-3.3.0a1.dev11+gac140c6.d20210423
  Running setup.py develop for message-ix
WARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617>
distutils: c:\users\fricko\anaconda3\Include\UNKNOWN
sysconfig: c:\users\fricko\anaconda3\Include
WARNING: Additional context:
user = False
home = None
root = None
prefix = None
WARNING: Ignoring invalid distribution -xmp (c:\users\fricko\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -pype1 (c:\users\fricko\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -xmp (c:\users\fricko\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -pype1 (c:\users\fricko\anaconda3\lib\site-packages)
Successfully installed message-ix-3.2.1.dev67+ga20ffb0.d20210427
WARNING: Ignoring invalid distribution -xmp (c:\users\fricko\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -pype1 (c:\users\fricko\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -xmp (c:\users\fricko\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -pype1 (c:\users\fricko\anaconda3\lib\site-packages)
```

**Comment by khaeru** [comment]
For context, this appears to be happening on:
- Windows
- Using Anaconda.

I also guess/assume:
- This is an install from source from a clone of the message_ix `master` branch.

Is that correct?

Finally, while these warning messages are generated, the install appears to succeed:
```
Successfully installed message-ix-3.2.1.dev67+ga20ffb0.d20210427
```

Can you confirm that message_ix is usable even after these messages appear?


First:
> `WARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617>`

The URL indicated says “Please upgrade to the newest pip release”. Can you do this, then give `pip --version` to confirm?

Second:
```
WARNING: Ignoring invalid distribution -xmp (c:\users\fricko\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -pype1 (c:\users\fricko\anaconda3\lib\site-packages)
```
These appeared to be corrupted versions of the names "ixmp" and "Jpype1", with the initial characters replaced by a hyphen ("-").
Can you please:
- Check the contents of "setup.cfg" in the directory from which you are installing? Do these strings occur?
- Check the directory indicated in the warning (c:\users\fricko\anaconda3\lib\site-packages) and see if there are any files/subdirectories with these names (e.g. "-xmp")?

**Comment by OFR-IIASA** [comment]
@khaeru Answers below. Also, the message ONLY appeared after upgrading "pip" (to version 21.1). I did the install just moments before the update of "pip" and there were no issues. Immediately after the install, the error messages occurred.


Can you confirm that message_ix is usable even after these messages appear?
- Yes, it is usable, but note that I have previously installed ixmp/jpype, therefore on a fresh installation, this may not be the case.

Check the contents of "setup.cfg" in the directory from which you are installing? Do these strings occur?
- No, the config files are identical to those in the master repository. 

Check the directory indicated in the warning (c:\users\fricko\anaconda3\lib\site-packages) and see if there are any files/subdirectories with these names (e.g. "-xmp")?
- `~xmp`
- `~pype`

This is an install from source from a clone of the message_ix master branch.
- correct

**Comment by khaeru** [comment]
> Check the directory indicated in the warning (c:\users\fricko\anaconda3\lib\site-packages) and see if there are any files/subdirectories with these names (e.g. "-xmp")?
>
> - `~xmp`
> - `~pype`

Can you please remove these and then see if anything changes? I'm not sure how they appeared on your system (again, seems to match the pattern some editors/tools use for creating temporarily files/directories) but they may be confusing Python's machinery.

**Comment by OFR-IIASA** [comment]
A quick update.

i have removed the directories and re-ran the 'pip' on a fresh message_ix/master branch. The following messages were still displayed.

```
WARNING: Value for scheme.headers does not match. Please report this to <https://github.com/pypa/pip/issues/9617>
distutils: c:\users\fricko\anaconda3\Include\UNKNOWN
sysconfig: c:\users\fricko\anaconda3\Include
WARNING: Additional context:
user = False
home = None
root = None
prefix = None
Obtaining file:///C:/repo/message_ix
```

There is though a new version of pip which was just released (12.1.1) and with this update, none of the observed warnings appear. The issue can therefore be closed.

```
$ pip install -e .
Obtaining file:///C:/repo/message_ix
Requirement already satisfied: click in c:\users\fricko\anaconda3\lib\site-packages (from message-ix==3.3.0a1.dev25+ga73cc36) (7.1.2)
Requirement already satisfied: ixmp>=3.2.0 in c:\repo\ixmp (from message-ix==3.3.0a1.dev25+ga73cc36) (3.2.1.dev80+g40fc589)
Requirement already satisfied: numpy in c:\users\fricko\anaconda3\lib\site-packages (from message-ix==3.3.0a1.dev25+ga73cc36) (1.19.2)
Requirement already satisfied: pandas in c:\users\fricko\anaconda3\lib\site-packages (from message-ix==3.3.0a1.dev25+ga73cc36) (1.1.5)
Requirement already satisfied: PyYAML in c:\users\fricko\anaconda3\lib\site-packages (from message-ix==3.3.0a1.dev25+ga73cc36) (5.3.1)
Requirement already satisfied: setuptools>=41 in c:\users\fricko\anaconda3\lib\site-packages (from message-ix==3.3.0a1.dev25+ga73cc36) (51.0.0.post20201207)
Requirement already satisfied: genno>=1.1.0 in c:\users\fricko\anaconda3\lib\site-packages (from ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (1.3.0)
Requirement already satisfied: JPype1>=1.2.1 in c:\users\fricko\anaconda3\lib\site-packages (from ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (1.2.1)
Requirement already satisfied: openpyxl in c:\users\fricko\anaconda3\lib\site-packages (from ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (3.0.5)
Requirement already satisfied: pint in c:\users\fricko\anaconda3\lib\site-packages (from ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (0.11)
Requirement already satisfied: xarray in c:\users\fricko\anaconda3\lib\site-packages (from ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (0.15.1)
Requirement already satisfied: dask[array]>=2.14 in c:\users\fricko\anaconda3\lib\site-packages (from genno>=1.1.0->ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (2020.12.0)
Requirement already satisfied: sparse>=0.12 in c:\users\fricko\anaconda3\lib\site-packages (from genno>=1.1.0->ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (0.12.0)
Requirement already satisfied: toolz>=0.8.2 in c:\users\fricko\anaconda3\lib\site-packages (from dask[array]>=2.14->genno>=1.1.0->ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (0.11.1)
Requirement already satisfied: typing-extensions in c:\users\fricko\anaconda3\lib\site-packages (from JPype1>=1.2.1->ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (3.7.4.3)
Requirement already satisfied: pytz>=2017.2 in c:\users\fricko\anaconda3\lib\site-packages (from pandas->message-ix==3.3.0a1.dev25+ga73cc36) (2020.4)
Requirement already satisfied: python-dateutil>=2.7.3 in c:\users\fricko\anaconda3\lib\site-packages (from pandas->message-ix==3.3.0a1.dev25+ga73cc36) (2.8.1)
Requirement already satisfied: six>=1.5 in c:\users\fricko\anaconda3\lib\site-packages (from python-dateutil>=2.7.3->pandas->message-ix==3.3.0a1.dev25+ga73cc36) (1.15.0)
Requirement already satisfied: scipy>=0.19 in c:\users\fricko\anaconda3\lib\site-packages (from sparse>=0.12->genno>=1.1.0->ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (1.5.2)
Requirement already satisfied: numba>=0.49 in c:\users\fricko\anaconda3\lib\site-packages (from sparse>=0.12->genno>=1.1.0->ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (0.51.2)
Requirement already satisfied: llvmlite<0.35,>=0.34.0.dev0 in c:\users\fricko\anaconda3\lib\site-packages (from numba>=0.49->sparse>=0.12->genno>=1.1.0->ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (0.34.0)
Requirement already satisfied: et-xmlfile in c:\users\fricko\anaconda3\lib\site-packages (from openpyxl->ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (1.0.1)
Requirement already satisfied: jdcal in c:\users\fricko\anaconda3\lib\site-packages (from openpyxl->ixmp>=3.2.0->message-ix==3.3.0a1.dev25+ga73cc36) (1.4.1)
Installing collected packages: message-ix
  Attempting uninstall: message-ix
    Found existing installation: message-ix 3.3.0a1.dev25+ga73cc36
    Uninstalling message-ix-3.3.0a1.dev25+ga73cc36:
      Successfully uninstalled message-ix-3.3.0a1.dev25+ga73cc36
  Running setup.py develop for message-ix
Successfully installed message-ix-3.3.0a1.dev25+ga73cc36`
```

**Comment by khaeru** [comment]
Thanks for the update!
